### ISSUE TITLE
Locked threads

### ISSUE THREAD

**Issue by SoniEx2** [mastodon/mastodon#16504]
### Pitch

Moderators should be able to lock threads. This should be unrelated to federation of such locked status and is meant as a moderation tool. Instances where a thread is locked ignore any notifications/replies related to the thread and prevent replying to it.

### Motivation

We think instances with more active moderation would be able to use such tools to prevent abuse and harassment, especially so-called "flame wars". In particular, if an user posts an inflammatory post, said post can be locked by the admins to break the feedback loop of post bad content -> get lots of attention -> post more bad content -> get more attention, without punishing the user.

**Comment by weex** [comment]
Makes sense to me but do you know of a specific actively-moderated instance that would use this feature? No need to name them, yes or no is fine.

**Comment by SoniEx2** [comment]
We don't know, we haven't been on the fediverse in a while. But we have been paying attention to how things go wrong elsewhere, and what can be done about it, and we think such a moderation tool would be a good fit for a self-proclaimed *ethical* alternative to those other systems.

**Comment by ThisIsMissEm** [comment]
In order for this to be ever implemented, we'd [need a FEP written](https://codeberg.org/fediverse/fep/src/branch/main/README.md#submitting-a-fep) for how this would work.

Though, there's a complicating factor in that Mastodon doesn't actually currently "manage" the replies collection, which causes a whole bunch of other problems; so when it comes to "locking a thread" the source server for the post that's locked would need to be actively rejecting replies.
